packages/rum-core/src/performance-monitoring/transaction.js

```diff
@@ -73,9 +73,16 @@ class Transaction extends SpanBase {
   }
 
   redefine(name, type, options) {
-    this.name = name
-    this.type = type
-    this.options = extend(this.options, options)
+    if (name) {
```

Comment:
All these checks are repetitive and unnecessary if done right - check the PR here - https://github.com/elastic/apm-agent-rum-js/pull/294/

Reply:
Here we don't have any default values, instead we are rejecting falsy values!

Reply:
Ideally there shouldn't be any falsy values. It will be just the defaults if its falsy. Just having this check everywhere in the codebase is not solving anything to be honest. We do default to `Unknown` and `custom` for name and type everywhere.

Reply:
Not sure how defaulting to `Unknown` and `custom`  is relevant here, but here we shouldn't set falsy values to these properties therefore either this method or it's caller should check for falsy values. The alternative is to have separate methods like `setName`, `setType` etc, don't think is any better. 

I'm open to suggestions though.